common/config/betterstack.go

```diff
@@ -0,0 +1,87 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/odigos-io/odigos/common"
+)
+
+const (
+	BetterStackMetricsEndpoint = "BETTERSTACK_METRICS_ENDPOINT"
+	BetterStackLogsEndpoint    = "BETTERSTACK_LOGS_ENDPOINT"
+)
+
+var (
+	ErrorBetterStackMetricsEndpointMissing = errors.New("BetterStack is missing a required field (\"BETTERSTACK_METRICS_ENDPOINT\"), BetterStack will not be configured")
+	ErrorBetterStackLogsEndpointMissing    = errors.New("BetterStack is missing a required field (\"BETTERSTACK_LOGS_ENDPOINT\"), BetterStack will not be configured")
+)
+
+type BetterStack struct{}
+
+func (j *BetterStack) DestType() common.DestinationType {
+	return common.BetterStackDestinationType
+}
+
+func (j *BetterStack) ModifyConfig(dest ExporterConfigurer, cfg *Config) error {
+	config := dest.GetConfig()
+	uniqueUri := "betterstack-" + dest.GetID()
+
+	processorName := "attributes/betterstack"
+	cfg.Processors[processorName] = GenericMap{
+		"actions": []GenericMap{
+			{
+				"key":    "better_stack_source_token",
+				"value":  "${BETTERSTACK_TOKEN}",
+				"action": "insert",
+			},
+		},
+	}
+
+	if isMetricsEnabled(dest) {
+		url, exists := config[BetterStackMetricsEndpoint]
+		if !exists {
+			return ErrorBetterStackMetricsEndpointMissing
+		}
+
+		endpoint, err := parseOtlpHttpEndpoint(url)
+		if err != nil {
+			return err
+		}
+
+		exporterName := "prometheusremotewrite/" + uniqueUri
+		cfg.Exporters[exporterName] = GenericMap{
+			"endpoint": endpoint,
+		}
+
+		pipeName := "metrics/" + uniqueUri
+		cfg.Service.Pipelines[pipeName] = Pipeline{
+			Processors: []string{processorName},
+			Exporters:  []string{exporterName},
+		}
+	}
+
+	if isLoggingEnabled(dest) {
+		url, exists := config[BetterStackLogsEndpoint]
+		if !exists {
+			return ErrorBetterStackLogsEndpointMissing
```

Comment:
If we return here, the config map will be partially applied (we can already inject the traces to the config but then return before setting up the metrics).

Consider first validating the data, and then either adding the destination as a whole, or not add it at all with an error